src/wasm/WasmModule.cpp

```diff
@@ -129,11 +177,13 @@ void WasmModule::restore(const std::string& snapshotKey)
         SPDLOG_DEBUG("Growing memory by {} bytes to restore snapshot",
                      bytesRequired);
         this->growMemory(bytesRequired);
-    } else {
+    } else if (data.size < memSize) {
         size_t shrinkBy = memSize - data.size;
         SPDLOG_DEBUG("Shrinking memory by {} bytes to restore snapshot",
                      shrinkBy);
         this->shrinkMemory(shrinkBy);
+    } else {
+        SPDLOG_DEBUG("Memory already correct size for snapshot ({})", memSize);
```

Comment:
This logging was unclear when the memory was the right size already; it would print `Shrinking memory by 0 bytes to restore snapshot`